testing/framework/test-framework.rst

```diff
@@ -48,29 +48,33 @@ There are three types of SCons tests:
 Contrasting end-to-end and unit tests
 -------------------------------------
 
-In general, functionality with end-to-end tests
-should be considered a hardened part of the public interface (that is,
-something that a user might do) and should not be broken.  Unit tests
-are now considered more malleable, more for testing internal interfaces
-that can change so long as we don't break users' ``SConscript`` files.
-(This wasn't always the case, and there's a lot of meaty code in many
-of the unit test scripts that does, in fact, capture external interface
+In general, end-to-end tests verify hardened parts of the public interface:
+interfaces documented in the manpage that a user might use in their
+project. These cannot be broken (of course, errors can be corrected,
+though sometimes a transition period will be required).
+Unit tests are now considered for testing internal interfaces, which do
+not themselves directly have API guarantees. Things can change here if
+necessary, as long as we don't break users' ``SConscript`` files.
+Note this strategy is new-ish, and there's a lot of meaty code in many
+of the unit test scripts that, in fact, captures external interface
 behavior.  In general, we should try to move those things to end-to-end
-scripts as we find them.)
+scripts as we find them.
```

Comment:
Not sure I agree with the assertion that testing of fixed APIs should be moved to system tests.

If unit tests can adequately test API, they should be used as they are considerably faster and simpler to debug.